iocore/net/UnixConnection.cc

```diff
@@ -320,14 +320,15 @@ Connection::connect(sockaddr const *target, NetVCOptions const &opt)
 
   int res;
 
-  this->setRemote(target);
+  if (target != NULL)
+    this->setRemote(target);
 
   // apply dynamic options with this.addr initialized
   apply_options(opt);
 
   cleaner<Connection> cleanup(this, &Connection::_cleanup); // mark for close until we succeed.
 
-  res = ::connect(fd, target, ats_ip_size(target));
+  res = ::connect(fd, &this->addr.sa, ats_ip_size(&this->addr.sa));
```

Comment:
@masaori335 con.connect() always using addr.sa as ::connect() target addr.
set addr.sa with target if target is not NULL.